assets/mods/simplify/postload.js

```diff
@@ -220,58 +222,63 @@
-	
-				//console.log("Replacing '" + settings.url + "' with '" + fullreplace[0]  + "'");
-	
-				if (fullreplace[0].indexOf('assets') === 0) {
-					settings.url = ig.root + fullreplace[0].substr(7);
-				} else {
-					settings.url = ig.root + fullreplace[0];
+			// Apply asset overrides.
+			settings.url = this._applyAssetOverrides(settings.url);
+
+			// To simplify the timeline, assume patching is always going on.
+			let patches = this._getAllAssetsEx(settings.url.substr(ig.root.length) + '.patch');
+
+			const success = settings.success;
+			let successArgs;
+			// Required 'loaded' == patchData.length + 1 (res.)
+			let loaded = 0;
+
+			const finalize = () => {
+				for (const entry of this.handlers) {
+					if(!entry.beforeCall && (!entry.filter || settings.url.substr(ig.root.length).match(entry.filter))) {
+						entry.handler(successArgs[0], settings.url.substr(ig.root.length));
+					}
 				}
-			}
-	
-			const patches = this._getAllAssets(settings.url.substr(ig.root.length) + '.patch');
-			if(patches && patches.length > 0){
-				const patchData = [];
-				const success = settings.success;
-				let successArgs;
-				let resourceLoaded = false;
-	
-				for (const patch of patches) {
-					this.loadJSON(patch)
-						.then(data => {
-							patchData.push(data);
-							if(patchData.length === patches.length && resourceLoaded){
-								this._applyPatches(successArgs[0], patchData);
-								success.apply(settings.context, successArgs);
-							}
-						})
-						.catch(err => {
-							console.error(err);
-							patchData.push({});
-						});
+				success.apply(settings.context, successArgs);
+			};
+
+			const check = () => {
+				if (loaded == patches.length + 1) {
+					// Done loading the main files, run the actual patches in sequence.
+					(async () => {
+						for (const patch of patches)
+							await this._applyPatch(successArgs[0], patch.data, patch.mod.baseDirectory);
+					})().then(() => {
+						finalize();
+					}).catch((err) => {
+						console.error(err);
+						finalize();
+					});
 				}
-	
-				settings.success = (...successArgs) => {
-					resourceLoaded = true;
-					if (patchData.length === patches.length) {
-						this._applyPatches(successArgs[0], patchData);
-	
-						for (const entry of this.handlers) {
-							if(!entry.beforeCall && (!entry.filter || settings.url.substr(ig.root.length).match(entry.filter))) {
-								entry.handler(successArgs[0], settings.url.substr(ig.root.length));
-							}
-						}
-	
-						success.apply(settings.context, successArgs);
-					}
-				};
+			};
+
+			// Start parallel requests. These call check() on completion. When it's all complete, finalize() occurs.
+
+			for (const patch of patches) {
+				this.loadJSON(patch.path)
+					.then(data => {
+						patch.data = data;
+						loaded++;
+						check();
+					})
+					.catch(err => {
+						console.error(err);
+						patch.data = {};
+						loaded++;
+						check();
+					});
```

Comment:
This entire logic can be simplified by iterating over patches, adding the promises to an array and using `Promise.all` to await the result. If order matters, a loop with `await` statement suffices.

Reply:
Ah, right, forgot that one